AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/hierarchical-state-machine-swift/StateAsyncSequence.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+class StateAsyncSequence<Element: Sendable>: AsyncSequence {
```

Comment:
Would it make sense to reuse [AmplifyAsyncSequence](https://github.com/aws-amplify/amplify-ios/blob/dev-preview/Amplify/Core/Support/AmplifyAsyncSequence.swift) here?

Reply:
Few things for not using AmplifyAsyncSequence:
- Do not want to rely on a public type here and depend on core, statemachine logic might be moved to a separate place in future
- We do not need most of the functionalities inside AmplifyAsyncSequence
- I kept the sequence as a `class` to help in identifying if it was dereferenced